src/Schemas.ts

```diff
@@ -510,6 +510,36 @@ export function union<TL, TR, SR>(left: Schema<TL, TL>, right: Schema<TR, SR>):
     );
 }
 
+type UnionDomainsReprs<Schemas extends unknown[], Acc = never> =
+    Schemas extends [infer H, ...(infer T)]
+        ? H extends Schema<infer D, infer D>
+            ? UnionDomainsReprs<T, Acc | D>
+            : never
+        : Acc;
+
+/**
+ * Construct a schema for a type union given any number of schemas. The schemas
+ * must be trivially encodable ('Schema<T, T>'). Like 'union' and 'unionOf',
+ * this is left-biased; the first (leftmost) schema will apply in the case
+ * where more than one schema would validate a value.
+ */
+export function multiUnion<
+    R extends NonEmptyArray<Schema<any, any>>
```

Comment:
`NonEmptyArray` here doesn't work; look at the type you get back if you say `const x = multiUnion();`. This would be extremely problematic. You can stick to an array here and settle for the resulting `Schema<never, never>`

Reply:
If the function ends up not being possible or annoying to write without a nonempty array (at a glance it would just require adding `aNever`, right? Not ideal but shouldn't actually do anything at runtime if it were at the right end of the union rather than the left somehow), let me know and I'll look harder at this. It might still be safe in context and I might just be wrong.

Reply:
Plus, I do like having the identity under unions in the library for algebraic purposes :p

Reply:
Upon further exploration I think the leak I found is a problem in typescript in general and not something you can address specifically by removing the `NonEmptyArray` here.

Whether or not you want to keep it is up to you! Apologies for the barrage of comments.